GraphWebApi/Telemetry/TelemetryProcessor.cs

```diff
@@ -0,0 +1,87 @@
+// ------------------------------------------------------------------------------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------------------------------------------------------------------------------
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using System;
+using System.Text.RegularExpressions;
+
+namespace GraphWebApi.Telemetry
+{
+    /// <summary>
+    /// Initializes a telemetry processor to sanitize http request urls to remove sensitive data.
+    /// </summary>
+    public class TelemetryProcessor : ITelemetryProcessor
+    {
+        private readonly ITelemetryProcessor _next;
+
+        private static Regex _guidRegex = new Regex(@"\b[A-F0-9]{8}(?:-[A-F0-9]{4}){3}-[A-F0-9]{12}\b",
+            RegexOptions.IgnoreCase | RegexOptions.Compiled);
+
+        // Matches patterns like users('MeganB@M365x214355.onmicrosoft.com')
+        private static Regex _emailRegex = new Regex(@"([a-zA-Z0-9_\.-]+)@([\da-zA-Z\.-]+)\.([a-zA-Z\.]{2,6})",
```

Comment:
With [C#9](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-9#fit-and-finish-features) you can omit the type in your constructor invocations: e.g.:

```suggestion
        private static readonly Regex _emailRegex = new(@"([a-zA-Z0-9_\.-]+)@([\da-zA-Z\.-]+)\.([a-zA-Z\.]{2,6})",
```
 Also, update the private fields to be `readonly`